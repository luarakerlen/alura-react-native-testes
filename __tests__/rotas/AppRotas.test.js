import React from 'react';
import { render } from '@testing-library/react-native';
import AppRotas from '../../src/rotas/AppRotas';
import Leilao from '../../src/telas/Leilao';
import ListaLeiloes from '../../src/telas/ListaLeiloes';
import { View } from 'react-native';

jest.mock('../../src/telas/Leilao');
jest.mock('../../src/telas/ListaLeiloes');

describe('rotas/AppRotas', () => {
	beforeAll(() => {
		Leilao.mockImplementation(() => <View testID='leilao-id' />);
		ListaLeiloes.mockImplementation(() => <View testID='lista-leiloes-id' />);
	});
	it('deve renderizar corretamente', () => {
		const { toJSON } = render(<AppRotas />);
		expect(toJSON()).toMatchSnapshot();
	});
});
