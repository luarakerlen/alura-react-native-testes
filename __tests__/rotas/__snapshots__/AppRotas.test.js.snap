// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rotas/AppRotas deve renderizar corretamente 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Lista de LeilÃµes"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          testID="lista-leiloes-id"
        />
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
